client/state/posts/reducer.js

```diff
@@ -282,6 +281,23 @@ export function edits( state = {}, action ) {
 				[ action.siteId ]: omit( state[ action.siteId ], action.postId || '' )
 			} );
 
+		case POST_SAVE_SUCCESS:
+			if ( ! state.hasOwnProperty( action.siteId ) || ! action.savedPost ) {
+				break;
+			}
+			const { postId, siteId, savedPost } = action;
+
+			if ( postId ) {
+				return state;
```

Comment:
Not entirely sure I follow here, if the prior `props.postId` was set, and it differs from `savedPost.ID`?  I hope that doesn't happen, but is that what you are talking about?